src/aolayer.cpp

```diff
@@ -290,7 +293,12 @@ void AOLayer::start_playback(QString p_image)
     frame = 0;
     continuous = false;
   }
-  // FIXME: this causes a hitch, find a more performant alternative
+  // CANTFIX: this causes a slight hitch
+  // The correct way of doing this would be to use QImageReader::jumpToImage() and 
+  // populate missing data in the movie ticker when it's needed. This is unforunately
+  // completely impossible, because QImageReader::jumpToImage() is not implemented
+  // in any image format AO2 is equipped to use. Instead, the default behavior is used - 
+  // that is, absolutely nothing.
```

Comment:
For emote continuity, perhaps layer data can be kept around if the animation had already been played back in full and it's likely it will be shown again (MRU cache), or we can try to load it asynchronously. Another idea is that seeking is generally a slow operation, but it shouldn't incur a massive cost as long as the number of copies done while reading irrelevant frames is minimized.

Reply:
We currently store every frame in a vector the first time the animation plays and fetch it from that vector on every subsequent play - this is as close to optimization as we can get, in my eyes.